source/base/quadrature.cc

```diff
@@ -284,6 +295,39 @@ Quadrature<dim>::memory_consumption () const
 }
 
 
+
+template <int dim>
+Quadrature<1>
+Quadrature<dim>::get_tensor_basis () const
+{
+  Assert (this->is_tensor_product_flag == true,
+          ExcMessage("This function only makes sense if this object represents a tensor product!"));
+
+  // Just take the first components of the first quadrature points
+  // and rescale the weights accordingly.
+  const unsigned int n_q_points_1d = std::round(std::pow(this->size(), 1./dim));
```

Comment:
`std::round` doesn't actually return an integer. I think you want to use a `static_cast<unsigned int>` here.